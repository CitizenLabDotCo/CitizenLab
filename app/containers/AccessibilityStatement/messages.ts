import { defineMessages } from 'react-intl';

export default defineMessages({
  // Cookie policy
  title: {
    id: 'app.containers.AccessibilityStatement.title',
    defaultMessage: 'Accessibility Statement',
  },
  pageDescription: {
    id: 'app.containers.AccessibilityStatement.pageDescription',
    defaultMessage: 'A statement on the accessibility of this website',
  },
  intro: {
    id: 'app.containers.AccessibilityStatement.intro',
    defaultMessage:
      'CitizenLab is committed to ensuring digital accessibility for people with disabilities. We are continually improving the user experience for everyone, and applying the relevant accessibility standards.',
  },
  changePreferencesText: {
    id: 'app.containers.AccessibilityStatement.changePreferencesText',
    defaultMessage: 'At any time, {changePreferencesButton}.',
  },
  changePreferencesButtonText: {
    id: 'app.containers.AccessibilityStatement.changePreferencesButtonText',
    defaultMessage: 'you can change your preferences',
  },
  effortsTitle: {
    id: 'app.containers.AccessibilityStatement.effortsTitle',
    defaultMessage: 'Efforts to support accessibility',
  },
  measures: {
    id: 'app.containers.AccessibilityStatement.measures',
    defaultMessage:
      'CitizenLab takes the following measures to ensure accessibility:',
  },
  measure1: {
    id: 'app.containers.AccessibilityStatement.measure1',
    defaultMessage: 'Accessibility is part of our mission statement.',
  },
  measure2: {
    id: 'app.containers.AccessibilityStatement.measure2',
    defaultMessage: 'An accessibility oicer or oicial has been appointed.',
  },
  measure3: {
    id: 'app.containers.AccessibilityStatement.measure3',
    defaultMessage:
      'Clear accessibility targets and responsibilities exist within the organization.',
  },
  conformanceStatus: {
    id: 'app.containers.AccessibilityStatement.conformanceStatus',
    defaultMessage: 'Conformance status',
  },
  currentStandard: {
    id: 'app.containers.AccessibilityStatement.currentStandard',
    defaultMessage: 'Current accessibility standard of the site:',
  },
  contentConformanceTitle: {
    id: 'app.containers.AccessibilityStatement.contentConformanceTitle',
    defaultMessage: 'Current content conformance status:',
  },
  contentConformanceInfo: {
    id: 'app.containers.AccessibilityStatement.contentConformanceInfo',
    defaultMessage:
      'Fully conformant: the standard content fully conforms to the accessibility standard without any exceptions.',
  },
  contentConformanceExceptions: {
    id: 'app.containers.AccessibilityStatement.contentConformanceExceptions',
    defaultMessage:
      'Users may add inaccessible content. It is possible that PDF or other file types including multimedia (user-generated content) are uploaded to the platform as attachments which are not accessible. Online workshops (live video) do not have subtitles.',
  },
  compatibilityTitle: {
    id: 'app.containers.AccessibilityStatement.compatibilityTitle',
    defaultMessage: 'Compatibility with browsers and assistive technology',
  },
  compatibilityBrowsers: {
    id: 'app.containers.AccessibilityStatement.compatibilityBrowsers',
    defaultMessage:
      'This site is designed to be fully compatible with the following browsers:',
  },
  browsers: {
    id: 'app.containers.AccessibilityStatement.browsers',
    defaultMessage: 'Google Chrome, version 80 and later',
  },
  compatibilityAssistiveTech: {
    id: 'app.containers.AccessibilityStatement.compatibilityAssistiveTech',
    defaultMessage:
      'This site is designed to be fully compatible with the following assistive technologies:',
  },
  screenReaderBugWarning: {
    id: 'app.containers.AccessibilityStatement.screenReaderBugWarning',
    defaultMessage:
      'Screen reader bugs may occur with different browser/screen reader combinations.',
  },
  technologiesTitle: {
    id: 'app.containers.AccessibilityStatement.technologiesTitle',
    defaultMessage: 'Technologies',
  },
  technologiesIntro: {
    id: 'app.containers.AccessibilityStatement.technologiesIntro',
    defaultMessage:
      'Accessibility of this site relies on the following technologies to work:',
  },
  assesmentMethodsTitle: {
    id: 'app.containers.AccessibilityStatement.assesmentMethodsTitle',
    defaultMessage: 'Assessment methods',
  },
  assesmentMethodsIntro: {
    id: 'app.containers.AccessibilityStatement.assesmentMethodsIntro',
    defaultMessage:
      'CitizenLab assessed the accessibility of this site using the following method(s):',
  },
  externalEvaluation: {
    id: 'app.containers.AccessibilityStatement.externalEvaluation',
    defaultMessage:
      'External evaluation: the site was evaluated by an external entity not involved in the design and development process.',
  },
  evaluationReportTitle: {
    id: 'app.containers.AccessibilityStatement.evaluationReportTitle',
    defaultMessage: 'Evaluation report',
  },
  evaluationReportIntro: {
    id: 'app.containers.AccessibilityStatement.evaluationReportIntro',
    defaultMessage: 'A recent evaluation report for the site can be found at:',
  },
  feedbackProcessTitle: {
    id: 'app.containers.AccessibilityStatement.feedbackProcessTitle',
    defaultMessage: 'Feedback process',
  },
  feedbackProcessIntro: {
    id: 'app.containers.AccessibilityStatement.feedbackProcessIntro',
    defaultMessage:
      'We welcome your feedback on the accessibility of this site. Please contact us via one of the following methods:',
  },
  email: {
    id: 'app.containers.AccessibilityStatement.email',
    defaultMessage: 'Email:',
  },
  postalAddress: {
    id: 'app.containers.AccessibilityStatement.postalAddress',
    defaultMessage: 'Postal address:',
  },
  citizenLabAddress: {
    id: 'app.containers.AccessibilityStatement.citizenLabAddress',
    defaultMessage: 'Boulevard Anspach 65, 1000 Brussels, Belgium',
  },
  responsiveness: {
    id: 'app.containers.AccessibilityStatement.responsiveness',
    defaultMessage: 'We aim to respond to feedback within 1-2 business days.',
  },
  informationPageName: {
    id: 'app.containers.AccessibilityStatement.informationPageName',
    defaultMessage: 'Information Page',
  },
  'terms-and-conditionsPageName': {
    id: 'app.containers.AccessibilityStatement.terms-and-conditionsPageName',
    defaultMessage: 'Terms & conditions',
  },
  'privacy-policyPageName': {
    id: 'app.containers.AccessibilityStatement.privacy-policyPageName',
    defaultMessage: 'Privacy Policy',
  },
  'accessibility-statementPageName': {
    id: 'app.containers.AccessibilityStatement.accessibility-statementPageName',
    defaultMessage: 'Accessibility statement',
  },
});
