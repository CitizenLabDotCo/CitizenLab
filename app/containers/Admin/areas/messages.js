/*
 * AdminPage.AreaDashboard Messages
 *
 * This contains all the text for the AdminPage.AreaDashboard component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  helmetTitle: {
    id: 'app.containers.AdminPage.AreaDashboard.helmetTitle',
    defaultMessage: 'Admin - areas dashboard',
  },
  helmetDescription: {
    id: 'app.containers.AdminPage.AreaDashboard.helmetDescription',
    defaultMessage: 'Area list in admin backoffice',
  },

});
