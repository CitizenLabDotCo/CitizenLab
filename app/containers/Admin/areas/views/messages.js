/*
 * AdminPage.DashboardPage Messages
 *
 * This contains all the text for the AdminPage.DashboardPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  headerIndex: {
    id: 'app.containers.AdminPage.Areas.Views.all.headerIndex',
    defaultMessage: 'All Areas',
  },
  deleteButton: {
    id: 'app.containers.AdminPage.Areas.Views.all.deleteButton',
    defaultMessage: 'Delete',
  },
  updateButton: {
    id: 'app.containers.AdminPage.Areas.Views.all.updateButton',
    defaultMessage: 'Edit',
  },
  createButton: {
    id: 'app.containers.AdminPage.Areas.Views.all.createButton',
    defaultMessage: 'New Area',
  },
  publishButton: {
    id: 'app.containers.AdminPage.Areas.Views.form.publishButton',
    defaultMessage: 'Publish Area',
  },

  areaLoadingMessage: {
    id: 'app.containers.AdminPage.Areas.Views.edit.areaLoadingMessage',
    defaultMessage: 'Loading Area...',
  },
  areaLoadingError: {
    id: 'app.containers.AdminPage.Areas.Views.edit.areaLoadingError',
    defaultMessage: 'Area not found!',
  },
  tableTitle: {
    id: 'app.containers.AdminPage.AreaDashboard.tableTitle',
    defaultMessage: 'Areas',
  },
});
