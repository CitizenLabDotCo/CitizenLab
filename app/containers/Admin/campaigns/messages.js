/*
 * AdminPage.CampaignsPage Messages
 *
 * This contains all the text for the admins email campaigns.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  listTitle: {
    id: 'app.containers.Admin.campaigns.listTitle',
    defaultMessage: 'Email Campaigns',
  },
  campaignDeletionConfirmation: {
    id: 'app.containers.Admin.campaigns.campaignDeletionConfirmation',
    defaultMessage: 'Are you sure?',
  },
  addCampaignButton: {
    id: 'app.containers.Admin.campaigns.addCampaignButton',
    defaultMessage: 'Create campaign',
  },
  deleteButtonLabel: {
    id: 'app.containers.Admin.campaigns.deleteButtonLabel',
    defaultMessage: 'Delete',
  },
  showButtonLabel: {
    id: 'app.containers.Admin.campaigns.showButtonLabel',
    defaultMessage: 'Details',
  },
  editButtonLabel: {
    id: 'app.containers.Admin.campaigns.editButtonLabel',
    defaultMessage: 'Edit',
  },
  manageButtonLabel: {
    id: 'app.containers.Admin.campaigns.manageButtonLabel',
    defaultMessage: 'Manage',
  },
  fieldSender: {
    id: 'app.containers.Admin.campaigns.fieldSender',
    defaultMessage: 'From',
  },
  fieldReplyTo: {
    id: 'app.containers.Admin.campaigns.fieldReplyTo',
    defaultMessage: 'Replies should go to',
  },
  fieldTo: {
    id: 'app.containers.Admin.campaigns.fieldTo',
    defaultMessage: 'To',
  },
  fieldSubject: {
    id: 'app.containers.Admin.campaigns.fieldSubject',
    defaultMessage: 'Email Subject',
  },
  fieldBody: {
    id: 'app.containers.Admin.campaigns.fieldBody',
    defaultMessage: 'Email Message',
  },
  addCampaignTitle: {
    id: 'app.containers.Admin.campaigns.addCampaignTitle',
    defaultMessage: 'New campaign',
  },
  formTitleWho: {
    id: 'app.containers.Admin.campaigns.formTitleWho',
    defaultMessage: 'Who',
  },
  formTitleWhat: {
    id: 'app.containers.Admin.campaigns.formTitleWhat',
    defaultMessage: 'What',
  },
  formTitleWhen: {
    id: 'app.containers.Admin.campaigns.formTitleWhen',
    defaultMessage: 'When',
  },
  editCampaignTitle: {
    id: 'app.containers.Admin.campaigns.editCampaignTitle',
    defaultMessage: 'Edit campaign',
  },
  draft: {
    id: 'app.containers.Admin.campaigns.draft',
    defaultMessage: 'Draft',
  },
  sent: {
    id: 'app.containers.Admin.campaigns.sent',
    defaultMessage: 'Sent',
  },
  sendNowButton: {
    id: 'app.containers.Admin.campaigns.sendNowButton',
    defaultMessage: 'Send now',
  },
  sendPreviewButton: {
    id: 'app.containers.Admin.campaigns.sendPreviewButton',
    defaultMessage: 'Send me a preview',
  },
  previewSentConfirmation: {
    id: 'app.containers.Admin.campaigns.previewSentConfirmation',
    defaultMessage: 'A preview email has been sent to your email address',
  },
});
