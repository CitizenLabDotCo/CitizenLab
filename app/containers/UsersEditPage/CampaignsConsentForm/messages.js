/*
 * CampaignsConsent Messages
 *
 * This contains all the text for the UsersEditPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  buttonSuccessLabel: {
    id: 'app.containers.CampaignsConsentForm.buttonSuccessLabel',
    defaultMessage: 'Success',
  },
  messageError: {
    id: 'app.containers.CampaignsConsentForm.messageError',
    defaultMessage: 'There was an error saving your email preferences.',
  },
  messageSuccess: {
    id: 'app.containers.CampaignsConsentForm.messageSuccess',
    defaultMessage: 'Your email preferences have been saved.',
  },
  /*
   * Notifications
   */
  h3: {
    id: 'app.containers.CampaignsConsentForm.h3',
    defaultMessage: 'Notifications',
  },
  h3sub: {
    id: 'app.containers.CampaignsConsentForm.h3sub',
    defaultMessage: 'Manage your notifications',
  },
  notifications_all_email: {
    id: 'app.containers.CampaignsConsentForm.notifications_all_email',
    defaultMessage: 'ALL EMAIL NOTIFICATIONS',
  },
  notifications_idea_post: {
    id: 'app.containers.CampaignsConsentForm.notifications_idea_post',
    defaultMessage: 'Someone posted an idea',
  },
  notifications_new_user: {
    id: 'app.containers.CampaignsConsentForm.notifications_new_user',
    defaultMessage: 'New user registration',
  },
  notifications_new_comments: {
    id: 'app.containers.CampaignsConsentForm.notifications_new_comments',
    defaultMessage: 'New comments',
  },
  notifications_all_app: {
    id: 'app.containers.CampaignsConsentForm.notifications_all_app',
    defaultMessage: 'ALL APP NOTIFICATIONS',
  },
  notifications_comment_on_comment: {
    id: 'app.containers.CampaignsConsentForm.notifications_comment_on_comment',
    defaultMessage: 'Someone commented on your comment',
  },
  notifications_mention: {
    id: 'app.containers.CampaignsConsentForm.notifications_mention',
    defaultMessage: 'Someone mentioned you',
  },
  notifications_idea_comment: {
    id: 'app.containers.CampaignsConsentForm.notifications_idea_comment',
    defaultMessage: 'Someone commented on your idea',
  },
  submit: {
    id: 'app.containers.CampaignsConsentForm.submit',
    defaultMessage: 'Save',
  },
  notificationsTitle: {
    id: 'app.containers.CampaignsConsentForm.notificationsTitle',
    defaultMessage: 'Notifications',
  },
  notificationsSubTitle: {
    id: 'app.containers.CampaignsConsentForm.notificationsSubTitle',
    defaultMessage: 'When do you want us to send you an email to notify you?',
  },
});
