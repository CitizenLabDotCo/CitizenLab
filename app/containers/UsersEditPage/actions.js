/*
 *
 * UsersEditPage actions
 *
 */

import {
  LOAD_CURRENT_USER_SUCCESS, LOAD_CURRENT_USER_ERROR, STORE_AVATAR_REQUEST,
  UPDATE_CURRENT_USER_SUCCESS, UPDATE_CURRENT_USER_ERROR, STORE_AVATAR_ERROR, UPDATE_USER_LOCALE,
  UPDATE_CURRENT_USER_REQUEST,
} from './constants';

export function loadCurrentUserSuccess(currentUser) {
  const validResponse = currentUser && currentUser.data;
  const currentUserWithAvatar = currentUser;

  if (!validResponse) {
    return {
      type: 'LOAD_CURRENT_USER_ERROR',
    };
  }

  currentUserWithAvatar.data.attributes.avatar =
    (validResponse && currentUserWithAvatar.data.attributes.avatar
      ? currentUserWithAvatar.data.attributes.avatar.medium
      : '');

  return {
    type: LOAD_CURRENT_USER_SUCCESS,
    payload: currentUserWithAvatar.data.attributes,
    userId: currentUserWithAvatar.data.id,
  };
}

export function loadCurrentUserError() {
  return {
    type: LOAD_CURRENT_USER_ERROR,
  };
}

export function updateCurrentUser(currentUser) {
  return {
    type: UPDATE_CURRENT_USER_REQUEST,
    payload: currentUser,
    userId: currentUser.userId,
  };
}

export function updateLocale(userLocale) {
  return {
    type: UPDATE_USER_LOCALE,
    userLocale,
  };
}

export function updateCurrentUserSuccess() {
  return {
    type: UPDATE_CURRENT_USER_SUCCESS,
  };
}

export function updateCurrentUserError(errors) {
  return {
    type: UPDATE_CURRENT_USER_ERROR,
    errors,
  };
}

export function storeAvatar(avatarBase64, userId) {
  return {
    type: STORE_AVATAR_REQUEST,
    avatarBase64,
    userId,
  };
}

export function storeAvatarError() {
  return {
    type: STORE_AVATAR_ERROR,
  };
}
