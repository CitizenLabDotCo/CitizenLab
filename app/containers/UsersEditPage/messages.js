/*
 * UsersEditPage Messages
 *
 * This contains all the text for the UsersEditPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  helmetTitle: {
    id: 'app.containers.UsersEditPage.helmetTitle',
    defaultMessage: 'User edit page',
  },
  helmetDescription: {
    id: 'app.containers.UsersEditPage.helmetDescription',
    defaultMessage: 'Edit user profile',
  },
  loading: {
    id: 'app.containers.UsersEditPage.loading',
    defaultMessage: 'Loading...',
  },
  loadError: {
    id: 'app.containers.UsersEditPage.loadError',
    defaultMessage: 'Can\'t load existing profile',
  },
  processing: {
    id: 'app.containers.UsersEditPage.processing',
    defaultMessage: 'Sending...',
  },
  buttonErrorLabel: {
    id: 'app.containers.UsersEditPage.buttonErrorLabel',
    defaultMessage: 'Error',
  },
  buttonSuccessLabel: {
    id: 'app.containers.UsersEditPage.buttonSuccessLabel',
    defaultMessage: 'Success',
  },
  messageError: {
    id: 'app.containers.UsersEditPage.messageError',
    defaultMessage: 'There was an error saving your profile.',
  },
  messageSuccess: {
    id: 'app.containers.UsersEditPage.messageSuccess',
    defaultMessage: 'Your profile has been saved.',
  },
  storeError: {
    id: 'app.containers.UsersEditPage.storeError',
    defaultMessage: 'Can\'t store updated profile',
  },
  stored: {
    id: 'app.containers.UsersEditPage.stored',
    defaultMessage: 'Profile stored',
  },
  /*
   * Basic information
  */
  h1: {
    id: 'app.containers.UsersEditPage.h1',
    defaultMessage: 'Basic information',
  },
  h1sub: {
    id: 'app.containers.UsersEditPage.h1sub',
    defaultMessage: 'Edit your basic information related to your account',
  },
  avatarUploadError: {
    id: 'app.containers.UsersEditPage.avatarUploadError',
    defaultMessage: 'Avatar upload failed',
  },
  firstName: {
    id: 'app.containers.UsersEditPage.firstName',
    defaultMessage: 'First name',
  },
  lastName: {
    id: 'app.containers.UsersEditPage.lastName',
    defaultMessage: 'Last name',
  },
  email: {
    id: 'app.containers.UsersEditPage.email',
    defaultMessage: 'E-mail address',
  },
  password: {
    id: 'app.containers.UsersEditPage.password',
    defaultMessage: 'Password',
  },
  language: {
    id: 'app.containers.UsersEditPage.language',
    defaultMessage: 'Language',
  },
  /*
   * Details
   */
  h2: {
    id: 'app.containers.UsersEditPage.h2',
    defaultMessage: 'Details',
  },
  h2sub: {
    id: 'app.containers.UsersEditPage.h2sub',
    defaultMessage: 'All information is private and help us to know you better',
  },
  gender: {
    id: 'app.containers.UsersEditPage.gender',
    defaultMessage: 'Gender',
  },
  male: {
    id: 'app.containers.UsersEditPage.male',
    defaultMessage: 'Male',
  },
  female: {
    id: 'app.containers.UsersEditPage.female',
    defaultMessage: 'Female',
  },
  unspecified: {
    id: 'app.containers.UsersEditPage.unspecified',
    defaultMessage: 'Unspecified',
  },
  bio: {
    id: 'app.containers.UsersEditPage.bio',
    defaultMessage: 'Bio',
  },
  bio_placeholder: {
    id: 'app.containers.UsersEditPage.bio_placeholder',
    defaultMessage: 'Write a short description of yourself',
  },
  domicile: {
    id: 'app.containers.UsersEditPage.domicile',
    defaultMessage: 'Domicile',
  },
  domicile_placeholder: {
    id: 'app.containers.UsersEditPage.domicile_placeholder',
    defaultMessage: 'Domicile',
  },
  outside: {
    id: 'app.containers.UsersEditPage.outside',
    defaultMessage: `{orgType, select,
      city {Outside of {name}}
      generic {None of these}
    }`,
  },
  birthdate: {
    id: 'app.containers.UsersEditPage.birthdate',
    defaultMessage: 'Date of Birth',
  },
  education: {
    id: 'app.containers.UsersEditPage.education',
    defaultMessage: 'Highest diploma',
  },
  education_placeholder: {
    id: 'app.containers.UsersEditPage.education_placeholder',
    defaultMessage: 'Highest diploma',
  },
  ISCED11_0: {
    id: 'app.containers.UsersEditPage.ISCED11_0',
    defaultMessage: 'E1',
  },
  ISCED11_1: {
    id: 'app.containers.UsersEditPage.ISCED11_1',
    defaultMessage: 'E2',
  },
  ISCED11_2: {
    id: 'app.containers.UsersEditPage.ISCED11_2',
    defaultMessage: 'E3',
  },
  ISCED11_3: {
    id: 'app.containers.UsersEditPage.ISCED11_3',
    defaultMessage: 'E4',
  },
  ISCED11_4: {
    id: 'app.containers.UsersEditPage.ISCED11_4',
    defaultMessage: 'E5',
  },
  ISCED11_5: {
    id: 'app.containers.UsersEditPage.ISCED11_5',
    defaultMessage: 'E6',
  },
  ISCED11_6: {
    id: 'app.containers.UsersEditPage.ISCED11_6',
    defaultMessage: 'E7',
  },
  ISCED11_7: {
    id: 'app.containers.UsersEditPage.ISCED11_7',
    defaultMessage: 'E8',
  },
  ISCED11_8: {
    id: 'app.containers.UsersEditPage.ISCED11_8',
    defaultMessage: 'E9',
  },
  /*
   * Notifications
   */
  h3: {
    id: 'app.containers.UsersEditPage.h3',
    defaultMessage: 'Notifications',
  },
  h3sub: {
    id: 'app.containers.UsersEditPage.h3sub',
    defaultMessage: 'Manage your notifications',
  },
  notifications_all_email: {
    id: 'app.containers.UsersEditPage.notifications_all_email',
    defaultMessage: 'ALL EMAIL NOTIFICATIONS',
  },
  notifications_idea_post: {
    id: 'app.containers.UsersEditPage.notifications_idea_post',
    defaultMessage: 'Someone posted an idea',
  },
  notifications_new_user: {
    id: 'app.containers.UsersEditPage.notifications_new_user',
    defaultMessage: 'New user registration',
  },
  notifications_new_comments: {
    id: 'app.containers.UsersEditPage.notifications_new_comments',
    defaultMessage: 'New comments',
  },
  notifications_all_app: {
    id: 'app.containers.UsersEditPage.notifications_all_app',
    defaultMessage: 'ALL APP NOTIFICATIONS',
  },
  notifications_comment_on_comment: {
    id: 'app.containers.UsersEditPage.notifications_comment_on_comment',
    defaultMessage: 'Someone commented on your comment',
  },
  notifications_mention: {
    id: 'app.containers.UsersEditPage.notifications_mention',
    defaultMessage: 'Someone mentioned you',
  },
  notifications_idea_comment: {
    id: 'app.containers.UsersEditPage.notifications_idea_comment',
    defaultMessage: 'Someone commented on your idea',
  },
  submit: {
    id: 'app.containers.UsersEditPage.submit',
    defaultMessage: 'Update profile',
  },
});
