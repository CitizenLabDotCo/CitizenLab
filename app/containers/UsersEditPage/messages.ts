/*
 * UsersEditPage Messages
 *
 * This contains all the text for the UsersEditPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  loading: {
    id: 'app.containers.UsersEditPage.loading',
    defaultMessage: 'Loading...',
  },
  invisibleTitleUserSettings: {
    id: 'app.containers.UsersEditPage.invisibleTitleUserSettings',
    defaultMessage: 'All settings for your account',
  },
  processing: {
    id: 'app.containers.UsersEditPage.processing',
    defaultMessage: 'Sending...',
  },
  buttonSuccessLabel: {
    id: 'app.containers.UsersEditPage.buttonSuccessLabel',
    defaultMessage: 'Success',
  },
  messageError: {
    id: 'app.containers.UsersEditPage.messageError',
    defaultMessage: 'There was an error saving your profile.',
  },
  messageSuccess: {
    id: 'app.containers.UsersEditPage.messageSuccess',
    defaultMessage: 'Your profile has been saved.',
  },
  /*
   * Basic information
  */
  h1: {
    id: 'app.containers.UsersEditPage.h1',
    defaultMessage: 'Basic information',
  },
  h1sub: {
    id: 'app.containers.UsersEditPage.h1sub',
    defaultMessage: 'Edit your basic information related to your account',
  },
  firstNames: {
    id: 'app.containers.UsersEditPage.firstNames',
    defaultMessage: 'First names',
  },
  lastName: {
    id: 'app.containers.UsersEditPage.lastName',
    defaultMessage: 'Last name',
  },
  email: {
    id: 'app.containers.UsersEditPage.email',
    defaultMessage: 'E-mail address',
  },
  password: {
    id: 'app.containers.UsersEditPage.password',
    defaultMessage: 'Password',
  },
  language: {
    id: 'app.containers.UsersEditPage.language',
    defaultMessage: 'Language',
  },
  /*
   * Deletion
   */
  deletionSection: {
    id: 'app.containers.UsersEditPage.deletionSection',
    defaultMessage: 'Delete your profile',
  },
  deletionSubtitle: {
    id: 'app.containers.UsersEditPage.deletionSubtitle',
    defaultMessage: 'This action can not be undone. The content you published on the platform will be anonymized. If you wish to delete all your content, you can contact us at support@citizenlab.co.',
  },
  deleteMyAccount: {
    id: 'app.containers.UsersEditPage.deleteMyAccount',
    defaultMessage: 'Delete my account',
  },
  deleteYourAccount: {
    id: 'app.containers.UsersEditPage.deleteYourAccount',
    defaultMessage: 'Delete your account',
  },
  deleteProfileError: {
    id: 'app.containers.UsersEditPage.deleteProfileError',
    defaultMessage: 'There was an issue deleting your profile, please try again later.',
  },
  logoAltText: {
    id: 'app.containers.UsersEditPage.logoAltText',
    defaultMessage: 'Logo of {tenantName}',
  },
  deleteAccountSubtext: {
    id: 'app.containers.UsersEditPage.deleteAccountSubtext',
    defaultMessage: 'We are sorry to see you go.',
  },
  reasonsToStayListTitle: {
    id: 'app.containers.UsersEditPage.reasonsToStayListTitle',
    defaultMessage: 'Before you go...',
  },
  tooManyEmails: {
    id: 'app.containers.UsersEditPage.tooManyEmails',
    defaultMessage: 'Receiving too many emails? You can manage your email preferences in your profile settings.',
  },
  privacyReasons: {
    id: 'app.containers.UsersEditPage.privacyReasons',
    defaultMessage: 'If you are worried with your privacy, have you read {conditionsLink}?',
  },
  conditionsLinkText: {
    id: 'app.containers.UsersEditPage.conditionsLinkText',
    defaultMessage: 'our conditions',
  },
  contactUs: {
    id: 'app.containers.UsersEditPage.contactUs',
    defaultMessage: 'You can reach out to explain what\'s not going well by clicking {feedbackLink}.',
  },
  feedbackLinkText: {
    id: 'app.containers.UsersEditPage.feedbackLinkText',
    defaultMessage: 'here',
  },
  feedbackLinkUrl: {
    id: 'app.containers.UsersEditPage.feedbackLinkUrl',
    defaultMessage: 'https://citizenlabco.typeform.com/to/z7baRP?source={url}',
  },
  noGoingBack: {
    id: 'app.containers.UsersEditPage.noGoingBack',
    defaultMessage: 'Once you click this button, we will have no way to restore your account.',
  },
  cancel: {
    id: 'app.containers.UsersEditPage.cancel',
    defaultMessage: 'Cancel',
  },
  /*
   * Details
   */
  gender: {
    id: 'app.containers.UsersEditPage.gender',
    defaultMessage: 'Gender',
  },
  bio: {
    id: 'app.containers.UsersEditPage.bio',
    defaultMessage: 'Bio',
  },
  bio_placeholder: {
    id: 'app.containers.UsersEditPage.bio_placeholder',
    defaultMessage: 'Write a short description of yourself',
  },
  domicile: {
    id: 'app.containers.UsersEditPage.domicile',
    defaultMessage: 'Domicile',
  },
  imageDropzonePlaceholder: {
    id: 'app.containers.UsersEditPage.imageDropzonePlaceholder',
    defaultMessage: 'Add your profile picture here (max. 5 Mb)',
  },
  a11y_imageDropzoneRemoveIconAriaTitle: {
    id: 'app.containers.UsersEditPage.a11y_imageDropzoneRemoveIconAriaTitle',
    defaultMessage: 'Remove profile picture',
  },
  outside: {
    id: 'app.containers.UsersEditPage.outside',
    defaultMessage: `{orgType, select,
      city {Outside of {name}}
      generic {None of these}
    }`,
  },
  submit: {
    id: 'app.containers.UsersEditPage.submit',
    defaultMessage: 'Update profile',
  },
  notificationsTitle: {
    id: 'app.containers.UsersEditPage.notificationsTitle',
    defaultMessage: 'Notifications',
  },
  notificationsSubTitle: {
    id: 'app.containers.UsersEditPage.notificationsSubTitle',
    defaultMessage: 'When do you want us to send you an email to notify you?',
  },
  user: {
    id: 'app.containers.UsersEditPage.user',
    defaultMessage: 'When do you want us to send you an email to notify you?',
  },
  /*
  * VerificationStatus
  */
  verifyTitle: {
    id: 'app.containers.UsersEditPage.verifyTitle',
    defaultMessage: 'Verify your identity',
  },
  verifyText: {
    id: 'app.containers.UsersEditPage.verifyText',
    defaultMessage: 'and become a verified citizen',
  },
  verifyNow: {
    id: 'app.containers.UsersEditPage.verifyNow',
    defaultMessage: 'Verify now',
  },
  verifiedTitle: {
    id: 'app.containers.UsersEditPage.verifiedTitle',
    defaultMessage: 'Your account has been verified',
  },
  verifiedText: {
    id: 'app.containers.UsersEditPage.verifiedText',
    defaultMessage: 'Your identity has been verified. Your city will be able to base decision on you as part of the community',
  },
  verifyAgain: {
    id: 'app.containers.UsersEditPage.verifyAgain',
    defaultMessage: 'Information about you are outdated ? Click here to verify again.',
  },
  blockedVerified: {
    id: 'app.containers.UsersEditPage.blockedVerified',
    defaultMessage: 'You can\'t edit this field because it contains verified informtion',
  },
  /*
  * Meta
  */
  metaTitle: {
    id: 'app.containers.UsersEditPage.metaTitle',
    defaultMessage: 'Profile settings page of {firstName} {lastName} | CitizenLab',
  },
  metaDescription: {
    id: 'app.containers.UsersEditPage.metaDescription',
    defaultMessage: 'This is the profile settings page of {firstName} {lastName} on the online participation platform of {tenantName}. Here you can verify your identity, edit your account information, delete your account and edit your email preferences.'
  },
});
