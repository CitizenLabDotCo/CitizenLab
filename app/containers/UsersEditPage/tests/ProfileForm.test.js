import React from 'react';
import { FormattedMessage } from 'utils/cl-intl';
import { createComponentWithIntl } from 'utils/testing/intl';

import messages from '../messages';
import { mapDispatchToProps } from '../index';
import { updateCurrentUser } from '../actions';

describe('<ProfileForm />', () => {
  describe('onFormSubmit', () => {
    it('updateCurrentUser action should be dispatched', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);

      const payload = {
        userId: 'anything',
      };

      result.updateCurrentUser(payload);

      expect(dispatch).toHaveBeenCalledWith(updateCurrentUser(payload));
    });
  });

  describe('<LabelInputPair />', () => {
    it('input\'s label should render correctly', () => {
      const id = 'first_name';
      const tree = createComponentWithIntl(
        <label htmlFor={id}>
          <FormattedMessage {...messages[id]} />
        </label>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
