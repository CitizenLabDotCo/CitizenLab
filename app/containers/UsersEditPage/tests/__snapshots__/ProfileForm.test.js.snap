// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileForm /> <LabelInputPair /> input's label should render correctly 1`] = `
<label
  htmlFor="first_name"
>
  <span>
    First Name
  </span>
</label>
`;
